### PR TITLE
zfs: 2.1.12 -> 2.1.13

### PR DESCRIPTION
**Pull request by adamcstephens**
## Description of changes

https://github.com/openzfs/zfs/releases/tag/zfs-2.1.13

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/stable.nix [hunk 1]
```diff
@@ -26,7 +27,7 @@ callPackage ./generic.nix args {
   ];
 
   # this package should point to the latest release.
-  version = "2.1.12";
+  version = "2.1.13";
```
Comment (Mic92) [inline pkgs/os-specific/linux/zfs/stable.nix:30]: Please also bump unstable to the same version.

File: pkgs/os-specific/linux/zfs/stable.nix [hunk 2]
```diff
@@ -14,10 +14,11 @@ callPackage ./generic.nix args {
   # check the release notes for compatible kernels
   kernelCompatible =
     if stdenv'.isx86_64 || removeLinuxDRM
-    then kernel.kernelOlder "6.4"
+    then kernel.kernelOlder "6.6"
     else kernel.kernelOlder "6.2";
-  latestCompatibleLinuxPackages = linuxKernel.packages.linux_6_1;
+  latestCompatibleLinuxPackages = linuxKernel.packages.linux_6_5;
```
Comment (Mic92) [inline pkgs/os-specific/linux/zfs/stable.nix:19]:
```suggestion
  latestCompatibleLinuxPackages = if stdenv'.isx86_64 || removeLinuxDRM
    then linuxKernel.packages.linux_6_4
    else linuxKernel.packages.linux_6_1;
```

File: pkgs/os-specific/linux/zfs/stable.nix (hunk 1 above)
Comment (Mic92) [reply pkgs/os-specific/linux/zfs/stable.nix:30]: nevermind. This would be a downgrade.

**Comment by toastal** [comment]
2.1.13 supports Linux 6.5 while 2.2.0-rc only supports Linux 6.4.

File: pkgs/os-specific/linux/zfs/stable.nix (hunk 2 above)
Comment (adamcstephens) [reply pkgs/os-specific/linux/zfs/stable.nix]: I understand this now that I looked up the module option for removeLinuxDRM. Interestingly #236459 didn't have this logic. Should we maybe just leave it in here moving forward even if 6.5 gets dropped before the next zfs version?

**Review by RaitoBezarius — APPROVED** [review]
LGTM.

**Comment by toastal** [comment]
Aside: how does one overlay the kernel modules? I’ve used commit as an overlay for `zfs` in Nixpkgs, but ran into issues compiling the kernel which is pulling in a version from the original Nixpkgs. I tried `pkgs.linuxPackages_x.extend (final: prev: { zfs = …; })`. This built, but that followed up with `extraModulePackages` missing `zfs`. Added, `zfs` but then got *this* error:

```
error: builder for '/nix/store/bhifa6fz4m2j7sgcifhzjb4dw2d1bsf5-linux-6.5.5-modules-shrunk.drv' failed with exit code 1;
       last 10 log lines:
       >   copying dependency: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/drivers/gpu/drm/amd/amdgpu/amdgpu.ko.xz
       > root module: kvm-amd
       >   copying dependency: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/virt/lib/irqbypass.ko.xz
       >   copying dependency: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/arch/x86/kvm/kvm.ko.xz
       >   dependency already copied: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/drivers/base/firmware_loader/firmware_class.ko.xz
       >   dependency already copied: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/drivers/char/hw_random/rng-core.ko.xz
       >   copying dependency: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/drivers/crypto/ccp/ccp.ko.xz
       >   copying dependency: /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5/kernel/arch/x86/kvm/kvm-amd.ko.xz
       > root module: zfs
       > modprobe: FATAL: Module zfs not found in directory /nix/store/q244lms4jggjab57mlvsgncdqmg5lgrh-linux-6.5.5-modules/lib/modules/6.5.5
       For full logs, run 'nix log /nix/store/bhifa6fz4m2j7sgcifhzjb4dw2d1bsf5-linux-6.5.5-modules-shrunk.drv'.
```

**Comment by victorklos** [comment]
Thanks! I can confirm that this works, rocking it now. A new AMD GPU on a system with ZFS is now possible :smile: 

Had to search a bit for what was needed to run this. For posterity:

    git clone -b the-pull-request-branch --depth 1 url-of-nixpkgs
    sudo nixos-rebuild -I nixpkgs=/path/to/cloned/nixpkgs switch

**Comment by RaitoBezarius** [comment]
> Thanks! I can confirm that this works, rocking it now. A new AMD GPU on a system with ZFS is now possible 😄
> 
> Had to search a bit for what was needed to run this. For posterity:
> 
> ```
> git clone -b the-pull-request-branch --depth 1 url-of-nixpkgs
> sudo nixos-rebuild -I nixpkgs=/path/to/cloned/nixpkgs switch
> ```

For posterity:
```
sudo nixos-rebuild -I nixpkgs="https://github.com/adamcstephens/nixpkgs/archive/94146957746e894891d3932423449c0d9d7577e0.tar.gz" switch
```

should probably also work ;).

File: pkgs/os-specific/linux/zfs/stable.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/os-specific/linux/zfs/stable.nix]: Yes.

**Comment by eadwu** [comment]
Is boot broken with this? Pool doesn't seem to be importing on my end.

**Comment by toastal** [comment]
Not sure if related, but I have

```
error: builder for '/nix/store/xlwrxh9ad4bfdn531z3kvz1qkcnn3jl7-kernel-modules.drv' failed with exit code 25;
       last 1 log lines:
       > error: collision between `/nix/store/w71j5py9q6wlpaqmnjpna07b6k2q1485-zfs-kernel-2.1.13-6.5.5/lib/modules/6.5.5/extra/spl/spl.ko.xz' and `/nix/store/3jl326y3mx07cf7nfb06y80r6j2jwq72-zfs-kernel-2.1.13-6.5.5/lib/modules/6.5.5/extra/spl/spl.ko.xz'
```

**Comment by shymega** [comment]
I can confirm that using the ZFS kernel module from `master` works on `linuxPackages_xanmod_stable`. However, I'm finding you have to compile the whole Xanmod kernel, manually.

I may switch to stock Linux whilst I wait for this bump to either hit unstable or 23.05. It wasn't working in 23.05 or unstable (Xanmod and ZFS) last time I tried, which was this evening.

Have to admit, I don't really like using Git/unstable for kernels, but with the lack of a similar construct like [this](https://nixos.wiki/wiki/ZFS#latestCompatibleLinuxPackages_of_ZFS_for_boot.kernelPackages) for Xanmod, I think I'll have to stick with it. Maybe lock Nixpkgs in my Flake.

**Pull request merged**